design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,485 @@
+    end
+
+    loop OSRM poll Release Service for new the OSProfiles peridically
+        OSRM->>Release Service: Retrieve EMT OSProfile tags
+        OSRM->>Inventory: List all EMT OSProfile available in the orchestrator
+        loop For each OSProfile in RS but not in Inventory
+            OSRM->>Inventory: Create EMT OSProfile
+        end    
+    end
+```
+
+## Affected components and Teams
+
+Edge Infrastructure Manager:
+- **Inventory (inv)**: changes to the schema 
+  1. OSProfile updates
+  2. Instance updates
+  3. New OSUpdatePolicy resource
+  4. New OSUpdateRun resource
+- **API**: API needs to be updated according to the schema changes.
+  1. OperatingSystem resource update
+  2. Instance resource update
+  3. New OSUpdatePolicy APIs (Create, Read, Delete)
+  4. New OSUpdateRun APIs (Read, Delete)
+- **Maintenance Manager (MM)**: The MM needs to be updated to handle the new
+  OSUpdatePolicy and OSUpdateRun resources.
+  1. Handle OSUpdatePolicy
+  2. Create and Update OSUpdateRun
+  3. Handle `os_updates_available` for Immutable OSes
+  4. Update southbound APIs to allow PUA to provide packages ready to be
+     updated, and handle `os_updates_available` for Mutable OSes
+  5. Update southbound APIs to allow PUA to provide the list of packages
+     installed on the EN (could be merged with point iv above).
+  6. Handle OS Update Run retention policy, and delete resources older than the
+     retention policy.
+- **OS Resource Manager (OSRM)**: OSRM adds support for new EMT versions, notice
+  of new updates for EMT, and fill package info from Ubuntu manifest.
+  1. Polling of new EMT versions from Release Service and create new OSProfiles
+     accordingly
+  2. Fetch Ubuntu Manifest referenced into the OSProfile from Release Service.
+     Parse it and fill the information into the OSProfile `installed_packages`
+     in Inventory.
+
+Edge Node:
+- **Platform Update Agent (PUA)**:
+  1. For mutable OSes, PUA needs to poll and check for new update to already
+     installed packages, and reports back to the MM the new versions available
+  2. Propagate installed packaged on the EN to the maintenance manager to
+     populate runtime packages on the Instance resource
+
+UI/CLI:
+1. Support for creation of OSUpdatePolicy via dedicated page to handle these
+   resources
+2. Show History of the Instance, via the OSUpdateRun resources
+3. Update to support the `os_updates_available` field in the Instance resource,
+   instead of using `desired_os` to support update badge
+4. Update to properly show runtime packages from the Instance instead of
+   OSProfile
+
+## Implementation plan
+
+The implementation plan is divided into 3 phases.
+
+### Phase 1 (for EMF 3.1 release)
+
+Goal: streamline day 2 workflows and support EN history. Keeping the notice of
+updates for EMT backward compatible, with MM that updates the Instance
+desired_os.
+
+1. Schema and REST API changes (inv.\*, API.\*) with deprecated fields
```

Comment:
shall we keep using the deprecated fields as well in case UI cannot support us?

Reply:
UI only consumes those fields, we can try to achieve this. I believe the main objective is to avoid feature regressions on the UI side.